### PR TITLE
Config: make config unmarshalling more consistent

### PR DESCRIPTION
**Pull request by reinkrul**
Previous PRs introduced harmful features:

- Allow engine config to have global config properties (e.g. `datadir`), while the global config is passed in `Configure()` (so it's redundant, leading to confusing code).
- Allow engines to share properties, e.g. auth engine using `network.certfile` which is confusing and leads to unwanted coupling and incorrect documentation.
- Engine config is flattened before unmarshal (allowing `koanf:"some.sub.path")`), while this isn't allowed in global config (which requires sub-structs), leading to hard-to-debug bugs.
- Config flattening leads to the weird construction where engine config needs to prefix their engine name in the Koanf tags (e.g. `network.enabletls`, which should just be `enabletls`).

The incentive of this change was an upcoming change which moves HTTP alternative bind configuration to a newly introduced HTTP engine, which couldn't get its map of interfaces definitions injected due to unwanted path flattening.

### CODE REVIEW

File: core/server_config.go [hunk 1]
```diff
@@ -203,6 +224,12 @@ func FlagSet() *pflag.FlagSet {
 	flagSet.StringSlice("http.default.cors.origin", nil, "When set, enables CORS from the specified origins for the on default HTTP interface.")
 	flagSet.String("tls.offload", "", "Whether to enable TLS offloading for incoming connections. If enabled `tls.certheader` must be configured as well.")
 	flagSet.String("tls.certheader", "", "Name of the HTTP header that will contain the client certificate when TLS is offloaded.")
+	// Legacy TLS settings, to be removed in v6:
+	flagSet.String("network.certfile", "", "PEM file containing the server certificate for the gRPC server. "+
+		"Required when `network.enabletls` is `true`.")
+	flagSet.String("network.certkeyfile", "", "PEM file containing the private key of the server certificate. "+
+		"Required when `network.enabletls` is `true`.")
+	flagSet.String("network.truststorefile", "truststore.pem", "PEM file containing the trusted CA certificates for authenticating remote gRPC servers.")
```
Comment (gerardsn) [inline core/server_config.go:232]: #1334 added a deprecation warning on these flags that are now lost

**Review by gerardsn — CHANGES_REQUESTED** [review]
It's hard to oversee the impact of these changes, but don't see anything weird

File: core/server_config.go [hunk 2]
```diff
@@ -61,6 +62,26 @@ type TLSConfig struct {
...
+	CertFile       string `koanf:"certfile"`
+	CertKeyFile    string `koanf:"certkeyfile"`
+	TrustStoreFile string `koanf:"truststorefile"`
+	// MaxCRLValidityDays defines the number of days a CRL can be outdated, after that it will hard-fail
+	MaxCRLValidityDays int `koanf:"maxcrlvaliditydays"`
+}
+
+// Enabled returns whether TLS is enabled according to this configuration.
+func (c NetworkTLSConfig) Enabled() bool {
+	return len(c.CertFile) > 0 || len(c.CertKeyFile) > 0
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply core/server_config.go:82]: It came from `Auth`, i'll remove it here and add it to auth again.

File: core/server_config.go (hunk 1 above)
Comment (reinkrul) [reply core/server_config.go:232]: Parallel PRs that are going to conflict either way, so I'll have to re-add them in this PR or the other (whichever gets merged last).

File: network/config.go [hunk 3]
```diff
@@ -19,44 +19,40 @@
...
+	GrpcAddr string `koanf:"grpcaddr"`
 	// ConnectionTimeout specifies the timeout before an outbound connection attempt times out (in milliseconds).
-	ConnectionTimeout int `koanf:"network.connectiontimeout"`
+	ConnectionTimeout int `koanf:"connectiontimeout"`
 	// MaxBackoff specifies the maximum backoff for outbound connections
-	MaxBackoff time.Duration `koanf:"network.maxbackoff"`
+	MaxBackoff time.Duration `koanf:"maxbackoff"`
 	// EnableTLS specifies whether to enable TLS for incoming connections.
-	EnableTLS bool `koanf:"network.enabletls"`
+	EnableTLS bool `koanf:"enabletls"`
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply network/config.go]: 👍

**Review by gerardsn — APPROVED** [review]

**Pull request merged**